### PR TITLE
MM-43904 - Fix: Calls: "Access to route for non-existent plugin" error log

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Customers are reporting many debug loglines with: ` Access to route for non-existent plugin       caller="app/plugin_requests.go:37" missing_plugin_id=com.mattermost.calls url=/plugins/com.mattermost.calls/config error="plugin not found: com.mattermost.calls"`
- This is because we were using an http request to `calls/config` and `calls/channels` as a check for whether the plugin is enabled. But this outputs the debug line above. 
- We needed a solution that would check if the plugin was enabled but wouldn't output the debug line. 
- Settled on pulling the publicly accessible plugin webapp manifests and checking if calls is registered there. If it is it's enabled.
- Bonus: We can use that to short-circuit some of the channel components (the calls banners), which will speed up the app.
- Also refactoring the plugin id while I'm at it.
- @enahum Since some customers are unhappy about this, would it make sense to put this into a dot-release for 1.51?

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-43904

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [ ] ~~Has UI changes~~
- [ ] ~~Includes text changes and localization file updates~~
- [ ] ~~Have tested against the 5 core themes to ensure consistency between them.~~

#### Device Information
- Android: 12, Pixel 6
- iOS: 15.3.1, iPhone 7 plus

#### Release Note
```release-note
Calls: Fixed "Access to route for non-existent plugin" in the server's log
```

### CODE REVIEW

**Comment by enahum** [comment]
> Since some customers are unhappy about this, would it make sense to put this into a dot-release for 1.51?

@amyblais

File: ios/Podfile.lock [hunk 1]
```diff
@@ -733,7 +733,7 @@ SPEC CHECKSUMS:
   FBLazyVector: 244195e30d63d7f564c55da4410b9a24e8fbceaa
   FBReactNativeSpec: c94002c1d93da3658f4d5119c6994d19961e3d52
   fmt: ff9d55029c625d3757ed641535fd4a75fedc7ce9
-  glog: 5337263514dd6f09803962437687240c5dc39aa4
+  glog: 85ecdd10ee8d8ec362ef519a6a45ff9aa27b2e85
```
Comment (enahum) [inline ios/Podfile.lock:736]: why the changes in this file ?

**Review by enahum — COMMENTED** [review]
LGTM, just want to make sure we revert the changes to the Podfile.lock

File: ios/Podfile.lock (hunk 1 above)
Comment (cpoile) [reply ios/Podfile.lock:736]: I don't know /shrug. Will revert

**Review by enahum — APPROVED** [review]

**Comment by amyblais** [comment]
@cpoile @streamer45 Let me know more on how severe/urgent the bug is. Optionally we could add this to v1.52 which is scheduled for May 16th. I'll add the v1.52 milestone for now, but we can discuss a dot release if you feel strongly about it.

**Comment by streamer45** [comment]
Thanks @amyblais 

The bug itself is harmless as it won't really affect the server. That said, it's particularly unexpected as installations are experiencing a lot of fail positive errors due to this issue causing alerts to go off and getting system admins alarmed/annoyed. A couple of very big customers have reported this internally already. 

My recommendation would be to include this if possible but with the understanding that this is not causing a direct impact to the functioning of the application so happy with any decision on this front.

**Review by streamer45 — APPROVED** [review]
Very nice work @cpoile , thanks for prioritizing this :raised_hands:

**Comment by amyblais** [comment]
> Thanks @amyblais
> 
> The bug itself is harmless as it won't really affect the server. That said, it's particularly unexpected as installations are experiencing a lot of fail positive errors due to this issue causing alerts to go off and getting system admins alarmed/annoyed. A couple of very big customers have reported this internally already.
> 
> My recommendation would be to include this if possible but with the understanding that this is not causing a direct impact to the functioning of the application so happy with any decision on this front.

Sounds good, feel free to cherry-pick to ``release-1.51`` and ``release-1.52`` branches @cpoile.

**Comment by enahum** [comment]
@cpoile this needs to be cherry-picked for 1.52 as well

**Comment by amyblais** [comment]
> @cpoile this needs to be cherry-picked for 1.52 as well

@enahum It was completed https://github.com/mattermost/mattermost-mobile/pull/6212.
